Frends.SFTP.UploadFiles/Frends.SFTP.UploadFiles/Definitions/RenamingPolicy.cs

```diff
@@ -1,270 +1,229 @@
-                filename = ReplaceSourceFileMacro(filename, originalFile);
+        return filename;
+    }
 
-            return filename;
-        }
+    private static string ExpandFileMasks(string filePath, string originalFileName)
+    {
+        string filename = filePath;
+        if (IsFileMask(filename))
+            filename = NameByMask(originalFileName, filename);
 
-        private string ExpandFileMasks(string filePath, string originalFileName)
-        {
-            string filename = filePath;
-            if (IsFileMask(filename))
-                filename = NameByMask(originalFileName, filename);
+        return filename;
+    }
 
-            return filename;
-        }
+    private static string NameByMask(string filename, string mask)
+    {
+        //remove extension if it is wanted to be changed, new extension is added later on to new filename
+        if (mask.Contains("*."))
+            if (Path.HasExtension(filename)) filename = Path.GetFileNameWithoutExtension(filename);
 
-        private static string NameByMask(string filename, string mask)
+        int i = mask.IndexOf("*");
+        if (i >= 0)
         {
-            //remove extension if it is wanted to be changed, new extension is added later on to new filename
-            if (mask.Contains("*."))
-            {
-                if (Path.HasExtension(filename))
-                    filename = Path.GetFileNameWithoutExtension(filename);
-            }
+            string tmp = mask.Substring(0, i);
+            return String.Concat(tmp + filename + mask.Substring(i + 1, (mask.Length - (i + 1))));
+        }
 
-            int i = mask.IndexOf("*");
-            if (i >= 0)
-            {
-                string tmp = mask.Substring(0, i);
-                return String.Concat(tmp + filename + mask.Substring(i + 1, (mask.Length - (i + 1))));
-            }
+        //Not an mask return mask.
+        return mask;
+    }
 
-            //Not an mask return mask.
-            return mask;
-        }
+    private static bool IsFileMacro(string s, IDictionary<string, Func<string, string>> macroDictionary)
+    {
+        if (s == null) return false;
 
-        private bool IsFileMacro(string s, IDictionary<string, Func<string, string>> macroDictionary)
+        foreach (var key in macroDictionary.Keys)
         {
-            if (s == null)
-            {
-                return false;
-            }
+            if (s.ToUpperInvariant().Contains(key.ToUpperInvariant())) return true;
+        }
 
-            foreach (var key in macroDictionary.Keys)
-            {
-                if (s.ToUpperInvariant().Contains(key.ToUpperInvariant()))
-                    return true;
-            }
+        return false;
+    }
 
-            return false;
-        }
+    private static bool IsFileMask(string s)
+    {
+        bool b = false;
+        if (s == null) return false;
+        if (s.Contains("*")) b = true;
```

Comment:
When searching for individual character, you should use quotes instead of double quotes. Otherwise compiler will think that you are searching for a string instead of a char.

Reply:
Fixed